src/daft-connect/src/op/execute/write.rs

```diff
@@ -0,0 +1,145 @@
+            }
+
+            let Ok(mode) = SaveMode::try_from(mode) else {
+                bail!("Invalid save mode: {mode}");
+            };
+
+            if !sort_column_names.is_empty() {
+                // todo(completeness): implement sort
+                warn!("Ignoring sort_column_names: {sort_column_names:?} (not yet implemented)");
+            }
+
+            if !partitioning_columns.is_empty() {
+                // todo(completeness): implement partitioning
+                warn!(
+                    "Ignoring partitioning_columns: {partitioning_columns:?} (not yet implemented)"
+                );
+            }
+
+            if let Some(bucket_by) = bucket_by {
+                // todo(completeness): implement bucketing
+                warn!("Ignoring bucket_by: {bucket_by:?} (not yet implemented)");
+            }
+
+            if !options.is_empty() {
+                // todo(completeness): implement options
+                warn!("Ignoring options: {options:?} (not yet implemented)");
+            }
+
+            if !clustering_columns.is_empty() {
+                // todo(completeness): implement clustering
+                warn!("Ignoring clustering_columns: {clustering_columns:?} (not yet implemented)");
+            }
+
+            match mode {
+                SaveMode::Unspecified => {}
+                SaveMode::Append => {}
+                SaveMode::Overwrite => {}
+                SaveMode::ErrorIfExists => {}
+                SaveMode::Ignore => {}
+            }
+
+            let Some(save_type) = save_type else {
+                bail!("Save type is required");
+            };
+
+            let path = match save_type {
+                SaveType::Path(path) => path,
+                SaveType::Table(table) => {
+                    let name = table.table_name;
+                    bail!("Tried to write to table {name} but it is not yet implemented. Try to write to a path instead.");
+                }
+            };
+
+            let mut plan = translation::to_logical_plan(input).await?;
+
+            plan.builder = plan
+                .builder
+                .table_write(&path, FileFormat::Parquet, None, None, None)
+                .wrap_err("Failed to create table write plan")?;
+
+            let optimized_plan = plan.builder.optimize()?;
+            let cfg = DaftExecutionConfig::default();
+            let native_executor = NativeExecutor::from_logical_plan_builder(&optimized_plan)?;
+            let mut result_stream = native_executor
+                .run(plan.psets, cfg.into(), None)?
+                .into_stream();
+
+            // this is so we make sure the operation is actually done
+            // before we return
+            //
+            // an example where this is important is if we write to a parquet file
+            // and then read immediately after, we need to wait for the write to finish
+            while let Some(_result) = result_stream.next().await {}
```

Comment:
Why not return the results of the write?

Reply:
because a write is not a read?

why would we want to do this?

Reply:
It returns the file names of the written files.

Reply:
https://github.com/Eventual-Inc/Daft/issues/3550

I want to unify these then. made a separate issue. can we resolve and merge? and then do that in another PR?

Reply:
Sounds good